compiler/rustc_type_ir/src/solve/inspect/format.rs

```diff
@@ -52,7 +52,10 @@ impl<'a, 'b> ProofTreeFormatter<'a, 'b> {
         writeln!(self.f, "}}")
     }
 
-    pub(super) fn format_goal_evaluation(&mut self, eval: &GoalEvaluation<'_>) -> std::fmt::Result {
+    pub(super) fn format_goal_evaluation<I: Interner>(
+        &mut self,
+        eval: &GoalEvaluation<I>,
```

Comment:
I feellike the proper approach here would be to use a `_marker: PhantomData<I>` on `ProofTreeFormatter` 🤔 this is a non-blocking

Reply:
Easy enough